### DISCUSSION TITLE
DNS PTR Record Management

### DISCUSSION THREAD

**Discussion by SwimGeek** [nautobot/nautobot#2667] [General]
Hi

In Nautobot IPAM, once you give an IP a DNS Name, is there some way to update your DNS servers automatically? 

Maybe a plugin that will do an API call to PowerDNS.

Please point me at any related docs.

Thank you.

**Comment by SwimGeek** [comment]
I found this:
https://github.com/candlerb/netbox-webhook-dnsupdate

Seems to use RFC2136 dynamic DNS updates, which is supported by PowerDNS.

Is there a Nautobot plugin that will do the same thing?

**Comment by SwimGeek** [comment]
This also looks useful:
https://github.com/sjm-steffann/netbox-ddns

**Comment by itdependsnetworks** [comment]
The pattern we see and most often use is based on ssot/diff sync. essentially we have built some tooling and several implementations to sync data between two systems. As an example, to cover the use case you are referring to, see this ssot plugin for infoblox  https://github.com/nautobot/nautobot-plugin-ssot-infoblox

**Reply by SwimGeek** [reply]
Hi. How would you suggest doing diff sync into PowerDNS? sync directly with PowerDNS API, or maybe go via PowerDNS-admin app?

**Reply by itdependsnetworks** [reply]
With no specific knowledge of powerdns, I would guess the api.

That being said, I would take a second to think of the philosophy of the ssot plugins, in that, many systems will need access to that data you have locally and many systems will not be using nautobot as the system of record/the place to write data. So as you think about how to actually scale an ssot to your network, how do you handle that? 

Point being, this is likely the first of many conceptually similar challenges to figure out, and I think the ssot/diff sync approach is key

**Reply by SwimGeek** [reply]
I'm considering just sticking with phpIPAM + PowedDNS (Mysql backend) - I find it really strange that with all the talk of 'Single Source of Truth', something as simple as naming an IP in one place is not a standard feature. The likes of Infoblox seem to get this right, but somehow in Netbox / Nautobot world people are happy to maintain this info in two places.  ¯\_(ツ)_/¯
